DataConnectors/AWS-CloudTrail-Ingestion-Lambda/IngestCloudTrailEventsToSentinel.ps1

```diff
@@ -0,0 +1,345 @@
+    $stringToHash = $method + "`n" + $contentLength + "`n" + $contentType + "`n" + $xHeaders + "`n" + $resource
+    $bytesToHash = [Text.Encoding]::UTF8.GetBytes($stringToHash)
+    $keyBytes = [Convert]::FromBase64String($sharedKey)
+    $sha256 = New-Object System.Security.Cryptography.HMACSHA256
+    $sha256.Key = $keyBytes
+    $calculatedHash = $sha256.ComputeHash($bytesToHash)
+    $encodedHash = [Convert]::ToBase64String($calculatedHash)
+    $authorization = 'SharedKey {0}:{1}' -f $customerId, $encodedHash
+    return $authorization
+}
+        
+# Function to create and post the request
+Function Invoke-LogAnalyticsData {
+    Param( 
+        $CustomerId, 
+        $SharedKey, 
+        $Body, 
+        $LogTable, 
+        $TimeStampField,
+        $resourceId)
+
+    $method = "POST"
+    $contentType = "application/json"
+    $resource = "/api/logs"
+    $rfc1123date = [DateTime]::UtcNow.ToString("r")
+    $contentLength = $Body.Length
+    $signature = New-BuildSignature `
+        -customerId $CustomerId `
+        -sharedKey $SharedKey `
+        -date $rfc1123date `
+        -contentLength $contentLength `
+        -method $method `
+        -contentType $contentType `
+        -resource $resource
+    $uri = "https://" + $CustomerId + ".ods.opinsights.azure.com" + $resource + "?api-version=2016-04-01"
+    $headers1 = @{
+        "Authorization"        = $signature;
+        "Log-Type"             = $LogTable;
+        "x-ms-date"            = $rfc1123date;
+        "x-ms-AzureResourceId" = $resourceId;
+        "time-generated-field" = $TimeStampField;
+    }  
+    $status = $false
+    do {
+        $response = Invoke-WebRequest -Uri $uri -Method $method -ContentType $contentType -Headers $headers1 -Body $Body
+        If ($reponse.StatusCode -eq 429) {
+            $rand = get-random -minimum 10 -Maximum 80
+            start-sleep -seconds $rand 
+        }
+        else { $status = $true }
+    }until($status) 
+    Remove-variable -name Body
+    return $response.StatusCode
+    
+}
+
+
+Function Ingest-Core-Fields-Single-Table {
+	Param(
+	$coreEvents)
+	
+	$coreJson = convertto-json $coreEvents -depth 5 -Compress    
+	$Table = "$LATableName" + "_All"
+	IF (($corejson.Length) -gt 28MB) {
+		Write-Host "Log length is greater than 28 MB, splitting and sending to Log Analytics"
+		$bits = [math]::Round(($corejson.length) / 20MB) + 1
+		$TotalRecords = $coreEvents.Count
+		$RecSetSize = [math]::Round($TotalRecords / $bits) + 1
+		$start = 0
+		For ($x = 0; $x -lt $bits; $X++) {
+			IF ( ($start + $recsetsize) -gt $TotalRecords) {
+				$finish = $totalRecords
+			}
+			ELSE {
+				$finish = $start + $RecSetSize
+			}
+			$body = Convertto-Json ($coreEvents[$start..$finish]) -Depth 5 -Compress
+			$result = Invoke-LogAnalyticsData -CustomerId $workspaceId -SharedKey $workspaceKey -Body $body -LogTable $Table -TimeStampField 'eventTime' -ResourceId $ResourceID          
+			if ($result -eq 200)
```

Comment:
what about other result type, 500/503/etc.  

I know for this basic function, you may not want to support resilient retry but at least log error code will be helpful. This will cause data loss for sure.